clients/src/main/java/org/apache/kafka/clients/consumer/internals/CommitRequestManager.java

```diff
@@ -827,26 +808,15 @@ abstract class RetriableRequestState extends RequestState {
          */
         abstract CompletableFuture<?> future();
 
-        /**
-         * Complete the request future with a TimeoutException if the request timeout has been
-         * reached, based on the provided current time.
-         */
-        void maybeExpire(long currentTimeMs) {
-            if (retryTimeoutExpired(currentTimeMs)) {
-                removeRequest();
-                isExpired = true;
-                future().completeExceptionally(new TimeoutException(requestDescription() +
-                    " could not complete before timeout expired."));
-            }
-        }
-
         /**
          * Build request with the given builder, including response handling logic.
          */
         NetworkClientDelegate.UnsentRequest buildRequestWithResponseHandling(final AbstractRequest.Builder<?> builder) {
             NetworkClientDelegate.UnsentRequest request = new NetworkClientDelegate.UnsentRequest(
                 builder,
-                coordinatorRequestManager.coordinator());
+                coordinatorRequestManager.coordinator(),
+                time.timer(requestTimeoutMs)
+            );
             request.whenComplete(
                 (response, throwable) -> {
                     long currentTimeMs = request.handler().completionTimeMs();
```

Comment:
I don't know 🤷‍♂️ 

Every so often I think about changing it, but I generally don't include changes that are unrelated to my PR.

Would you like me to change it in this PR?

Reply:
agree that it's confusing, but for the record, I guess the current in the name may come from the point of view that here's the moment a request completes, and we retrieve the completion time, so could I could see it as the current because of where it's called (but still +1 for better name, simply `completionTimeMs`, that btw aligns with the `handleClientResponse` func param where it's used right below)

Reply:
I will let you decide what to do. It just caught me.

Reply:
Can we handle this as a separate Jira or even a `[MINOR]` pull request?

Reply:
Sure!

Reply:
`'s/currentTimeMs/completionTimeMs/`